dataikuapi/dss/ml.py

```diff
@@ -308,50 +334,757 @@ def set_metric(self, metric=None, custom_metric=None, custom_metric_greater_is_b
+            message += "Expected a member of " + str(hyperparameter_names)
+            raise ValueError(message)
+
+    def _check_hyperparameter_searchable(self, hyperparameter_name):
+        assert isinstance(self[hyperparameter_name], dict), "Hyperparameter \"{}\" cannot be searched".format(hyperparameter_name)
+
+    def _check_hyperparameter_numerical(self, hyperparameter_name):
+        assert "range" in self[hyperparameter_name], "Hyperparameter \"{}\" is not numerical".format(hyperparameter_name)
+
+    def _check_range_bound(self, hyperparameter_name, range_bound):
+        assert "range" in self[hyperparameter_name], "Cannot modify range of non-numerical hyperparameter " \
+                                                         "\"{}\"".format(hyperparameter_name)
+        assert isinstance(range_bound, int) or isinstance(range_bound, float), \
+            "Invalid input type for hyperparameter \"{}\": ".format(hyperparameter_name) + \
+            "range bounds must be numbers"
+
+
+class RandomForestSettings(AlgorithmSettings):
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(RandomForestSettings, self).__init__(raw_settings, hyperparameter_search_params)
+        self.n_estimators = NumericalHyperparameterSettings("n_estimators", self)
+        self.min_samples_leaf = NumericalHyperparameterSettings("n_estimators", self)
+        self.max_tree_depth = NumericalHyperparameterSettings("max_tree_depth", self)
+        self.max_feature_prop = NumericalHyperparameterSettings("max_feature_prop", self)
+        self.max_features = NumericalHyperparameterSettings("max_features", self)
+        self.n_jobs = SingleValuedHyperparameterSettings("n_jobs", self, accepted_types=[int])
+        self.selection_mode = SingleValuedHyperparameterSettings("selection_mode", self, accepted_values=["auto", "sqrt", "log2", "number","prop"])
+
+
+class XGBoostSettings(AlgorithmSettings):
+
+    def __init__(self, raw_settings, hyperparameter_search_params):
+        super(XGBoostSettings, self).__init__(raw_settings, hyperparameter_search_params)
+        self.max_depth = NumericalHyperparameterSettings("max_depth", self)
+        self.learning_rate = NumericalHyperparameterSettings("learning_rate", self)
+        self.gamma = NumericalHyperparameterSettings("gamma", self)
+        self.min_child_weight = NumericalHyperparameterSettings("min_child_weight", self)
+        self.max_delta_step = NumericalHyperparameterSettings("max_delta_step", self)
+        self.subsample = NumericalHyperparameterSettings("subsample", self)
+        self.colsample_bytree = NumericalHyperparameterSettings("colsample_bytree", self)
+        self.colsample_bylevel = NumericalHyperparameterSettings("colsample_bylevel", self)
+        self.alpha = NumericalHyperparameterSettings("alpha", self)
+        self.lambda_ = NumericalHyperparameterSettings("lambda", self)
```

Comment:
why the `_` at the end of `lambda_` ?

Reply:
`lambda` is also a python keyword

Reply:
Actually getting a `SyntaxError` with the reserved keyword (Python 3.6.8).